nifi/nifi-nar-bundles/nifi-flume-bundle/nifi-flume-processors/src/main/java/org/apache/nifi/processors/flume/FlumeSourceProcessor.java

```diff
@@ -0,0 +1,191 @@
+            .defaultValue("")
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final Relationship SUCCESS = new Relationship.Builder().name("success").build();
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        this.descriptors = ImmutableList.of(SOURCE_TYPE, AGENT_NAME, SOURCE_NAME, FLUME_CONFIG);
+        this.relationships = ImmutableSet.of(SUCCESS);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @OnScheduled
+    public void onScheduled(final SchedulingContext context) {
+        try {
+            source = SOURCE_FACTORY.create(
+                    context.getProperty(SOURCE_NAME).getValue(),
+                    context.getProperty(SOURCE_TYPE).getValue());
+
+            String flumeConfig = context.getProperty(FLUME_CONFIG).getValue();
+            String agentName = context.getProperty(AGENT_NAME).getValue();
+            String sourceName = context.getProperty(SOURCE_NAME).getValue();
+            Configurables.configure(source,
+                    getFlumeSourceContext(flumeConfig, agentName, sourceName));
+
+            if (source instanceof EventDrivenSource) {
+                runner = new EventDrivenSourceRunner();
+                channel = new MemoryChannel();
+                Configurables.configure(channel, new Context());
+                channel.start();
+                source.setChannelProcessor(new ChannelProcessor(new NifiChannelSelector(channel)));
+                runner.setSource(source);
+                runner.start();
+            }
+        } catch (Throwable th) {
+            getLogger().error("Error creating source", th);
+            throw Throwables.propagate(th);
+        }
+    }
+
+    @OnUnscheduled
+    public void unScheduled() {
+        if (runner != null) {
+            runner.stop();
+        }
+        if (channel != null) {
+            channel.stop();
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context,
+            final ProcessSession session) throws ProcessException {
+        if (source instanceof EventDrivenSource) {
+            onEventDrivenTrigger(context, session);
+        } else if (source instanceof PollableSource) {
+            onPollableTrigger((PollableSource) source, context, session);
+        }
+    }
+
+    public void onPollableTrigger(final PollableSource pollableSource,
+            final ProcessContext context, final ProcessSession session)
+            throws ProcessException {
+        try {
+            pollableSource.setChannelProcessor(new ChannelProcessor(
+                    new NifiChannelSelector(new NifiChannel(session, SUCCESS))));
```

Comment:
I think the channel, selector, and processor can be reused by setting the session on the channel each time. The current session would be used to create transactions, which we don't expect to be long lived because this is a source. Then you might not have to start and stop the source each time, right?